### DISCUSSION TITLE
Safe oneliners!

### DISCUSSION THREAD

**Discussion by redraw** [saulpw/visidata#1875] [Show and tell]
I had a `domain` and `zone` columns, for which I wanted to create a new one from a expression that lookup their IPs. 

I found out this lib: [result](https://github.com/rustedpy/result), so that I could wrap the expression and unwrap it with some default  value if it fails. I would import `socket, result` from `System > Python > import library` and then create the column,

```
=result.as_result(Exception)(socket.gethostbyname)(f"{domain}.{zone}").unwrap_or("")
```

Then, I would freeze the column (`'`) and delete the expression column, so that it gets cached and the whole sheet doesn't get super slow to navigate.

**Comment by frosencrantz** [comment]
Thanks for sharing @redraw.


You might be able to remove the need for the freeze column if you are willing to add some code to you .visidatarc or create a simple plugin.   The shell column only runs the command for a row once.  You can look at what Visidata does for the ColumnShell column.  It does caching.  You can see it here:

https://github.com/saulpw/visidata/blob/develop/visidata/shell.py#L59

**Reply by redraw** [reply]
Great, it's just that the experience is different, and in this case it's not a column from a shell cmd, but a Python expression. 

Without freezing, visidata gets stucked until results are returned, meanwhile while freezing you can navigate rows and missing values are shown empty (Φ). This also happens just after I create the column from the Python expression. It gets stucked until the visible rows are completed.

I'm running the latest version, visidata==2.11, haven't looked at the code yet to debug why it might happen
